src/error/mod.rs

```diff
@@ -94,7 +94,7 @@ impl<'a> Ord for ErrorMessage<'a> {
 
 impl<'a> PartialOrd for ErrorMessage<'a> {
     fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
-        self.location.partial_cmp(&other.location)
+        Some(self.cmp(other))
```

Comment:
This should be the only change in code logic. Seems reasonable to me, but I don't know if there was a reason to originally do it in a non standard way